src/pyqasm/visitor.py

```diff
@@ -1516,9 +1517,40 @@ def _visit_constant_declaration(
                 error_node=statement,
                 span=statement.span,
             )
+
+        if statement.init_expression:
```

Comment:
Let's move this to a `validator` inside the `pulse/` subpackage. Essentially just mirror the directory structure that we have setup in PR #229